core/crawler.py

```
@@ -1,70 +1,67 @@
+        Args:
+            url (str): URL to fetch and parse
+            
+        Returns:
+            tuple: (html_content, links_found)
+                - html_content (str): Page HTML content
+                - links_found (list): List of absolute URLs found
+        """
+
+        self.links = []
+        self.baseUrl = url
+        response = urlopen(url)
```

Ruff (0.15.9)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)